doc/netplan.md

```diff
@@ -340,6 +340,90 @@ These options are available for all types of interfaces.
      :    Match this policy rule based on the type of service number applied to
           the traffic.
 
+## Authentication
+Netplan supports advanced authentication settings for ethernet and wifi
+interfaces, as well as individual wifi networks, by means of the ``auth`` block.
+
+``auth`` (mapping)
+
+:    Specifies authentication settings for a device of type ``ethernets:`` or
+     ``wifis:``, or a single wifi ``access-points:`` entry. If ``auth`` blocks
+     are specified on both a wifi device and one of its access points, only the
+     settings in the access point's ``auth`` block are taken into account;
+     however, the ``auth`` block of the wifi device, if present, is used for any
+     ``access-points:`` without their own ``auth`` block.
+
+     Example:
+
+          wifis:
+            wlp1s0:
+              auth:
+                key-mgmt: wpa-psk
+                psk: "d3f4ul7"
+              access-points:
+                office:
+                  auth:
+                    # the authentication settings above are ignored for the
+                    # wifi named "office"
+                    key-mgmt: wpa-eap
+                    eap-method: ttls
+                    anonymous-identity: "@internal.example.com"
+                    identity: "joe@internal.example.com"
+                    password: "v4ryS3kr1t"
+                home:
+                  mode: infrastructure
+                  # the authentication settings above are applied to the wifi
+                  # named "home"
+
+      The ``auth`` block supports the following properties:
+
+      ``key-mgmt`` (scalar)
```

Comment:
Is there anything more verbose  / meaningful to use than key-mgmt? Can we longform this to "key-management"? What about "protocol"?

Reply:
*protocol* is apparently used to mean the choice between [WPA and WPA2](https://w1.fi/cgit/hostap/tree/wpa_supplicant/wpa_supplicant.conf?id=0fe3ede0a4f6fe984bbaad90f1a9d9e8decc155d#n865), so overloading that might be risky. The long form (*management*) is probably okay, although the longer the word, the greater the potential for typos.

Reply:
I agree with cyphermox that 'key-management' is better than 'key-mgmt'.  Yes there's more chance for typos, but it has the advantage of being spell-checkable in wetware by english speakers, and it makes the meaning clear.

See also: creat(2).